api/encryption-handlers.go

```diff
@@ -661,76 +924,237 @@ func createOrReplaceKesConfigurationSecrets(ctx context.Context, clientSet K8sCl
+					ClientID:     encryptionCfg.Gcp.Secretmanager.Credentials.ClientID,
+					PrivateKeyID: encryptionCfg.Gcp.Secretmanager.Credentials.PrivateKeyID,
+					PrivateKey:   encryptionCfg.Gcp.Secretmanager.Credentials.PrivateKey,
+				}
+			}
+		}
+	case encryptionCfg.Azure != nil:
+		// Initialize Azure
+		kesConfig.Keystore.Azure = &kes.Azure{
+			KeyVault: &kes.AzureKeyVault{},
+		}
+		if encryptionCfg.Azure.Keyvault != nil {
+			kesConfig.Keystore.Azure.KeyVault.Endpoint = *encryptionCfg.Azure.Keyvault.Endpoint
+			if encryptionCfg.Azure.Keyvault.Credentials != nil {
+				kesConfig.Keystore.Azure.KeyVault.Credentials = &kes.AzureCredentials{
+					TenantID:     *encryptionCfg.Azure.Keyvault.Credentials.TenantID,
+					ClientID:     *encryptionCfg.Azure.Keyvault.Credentials.ClientID,
+					ClientSecret: *encryptionCfg.Azure.Keyvault.Credentials.ClientSecret,
+				}
+			}
 		}
 	}
+	return kesConfig.Marshal()
+}
 
-	var serverRawConfig []byte
-	var err error
+// getKesConfigMethod identify the config method to use based from the KES image name
+func getKesConfigMethod(image string) (configVersion, error) {
+	version, err := getKesConfigVersion(image)
+	if err != nil {
+		return nil, err
+	}
+	// switch for future (or previous) versions of KES config
+	switch version {
+	case KesConfigVersion1:
+		return createKesConfigV1, nil
+	default:
+		return createKesConfigV2, nil
+	}
+}
 
-	if encryptionCfg.Raw != "" {
-		serverRawConfig = []byte(encryptionCfg.Raw)
-		// verify provided configuration is in valid YAML format
-		var configTest kes.ServerConfig
-		err = yaml.Unmarshal(serverRawConfig, &configTest)
-		if err != nil {
-			return nil, nil, err
-		}
+func getKesConfigVersion(image string) (string, error) {
+	version := KesConfigVersion2
+
+	imageStrings := strings.Split(image, ":")
+	var imageTag string
+	if len(imageStrings) > 1 {
+		imageTag = imageStrings[1]
 	} else {
-		// Generate Yaml kesConfiguration for KES
-		serverRawConfig, err = yaml.Marshal(kesConfig)
-		if err != nil {
-			return nil, nil, err
+		return "", fmt.Errorf("%s not a valid KES release tag", image)
+	}
+
+	if imageTag == "edge" {
+		return KesConfigVersion2, nil
+	}
+
+	if imageTag == "latest" {
+		return KesConfigVersion2, nil
+	}
+
+	// When the image tag is semantic version is config v1
+	if semver.IsValid(imageTag) {
+		// Admin is required starting version v0.22.0
+		if semver.Compare(imageTag, "v0.22.0") < 0 {
+			return KesConfigVersion1, nil
 		}
+		return KesConfigVersion2, nil
 	}
 
-	// delete KES configuration secret if exists
-	if err := clientSet.deleteSecret(ctx, ns, kesConfigurationSecretName, metav1.DeleteOptions{}); err != nil {
-		// log the errors if any and continue
-		LogError("deleting secret name %s failed: %v, continuing..", kesConfigurationSecretName, err)
+	releaseTagNoArch := imageTag
+	fields := strings.Split(imageTag, "-")
+	// here we will remove the arch suffix if present, ie: 2023-05-02T22-48-10Z-arm64
+	if len(fields) > 5 {
```

Comment:
could you please remove this magic number and make it a constant? to know better what that means (it will self document)

Reply:
what about using a regex to identify if a tag matches the format with architecture in it? I think it is way more self describing that split by dashes

```go
kesImageTagWithArchRegexPattern = `(\d{4}-\d{2}-\d{2}T\d{2}-\d{2}-\d{2}Z)(-.*)`
```

Reply:
That would be a much more robust implementation I agree!